src/test/java/WebDriverScenarioWithTests.java

```diff
@@ -0,0 +1,122 @@
+import org.openqa.selenium.By;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.support.ui.ExpectedCondition;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import ru.yandex.qatools.ashot.AShot;
+import ru.yandex.qatools.ashot.Screenshot;
+import ru.yandex.qatools.ashot.shooting.ShootingStrategies;
+
+import javax.imageio.ImageIO;
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.logging.Logger;
+
+
+public class WebDriverScenarioWithTests {
+
+        private WebDriver driver;
+        Logger logger = Logger.getGlobal();
+
+        @BeforeMethod(alwaysRun = true)
+        public void browserSetup() {
+        driver = new ChromeDriver();
+    }
+
+        @Test(description = "JIRA binding can be here")
+        public void commonFlow() throws IOException {
+
+            ExpectedCondition<Boolean> pageLoadCondition = new
+                    ExpectedCondition<Boolean>() {
+                        public Boolean apply(WebDriver driver) {
+                            return ((JavascriptExecutor) driver).executeScript("return document.readyState").equals("complete");
+                        }
+                    };
+            WebDriverWait wait = new WebDriverWait(driver, 10);
+
+        driver.get("http://google.com");
+        driver.manage().window().maximize();
+        WebElement searchInput = driver.findElement(By.name("q"));
+        searchInput.sendKeys("wikipedia");
+        new WebDriverWait(driver, 20).until(ExpectedConditions
+                .elementToBeClickable(By.xpath("//input[@name='btnK']"))).click();
+        List<WebElement> searchResults = driver.findElements(By.xpath("//cite[contains(text(),'wikipedia')]"));
+        logger.info(String.format("Search results number for requested text: %s", searchResults.size()));
+
+        //3. Validation
+        WebElement firstElementOfSearchResults = driver.findElement(By.xpath("//cite[@class='iUh30 Zu0yb qLRx3b tjvcx']"));
+        String contentOfFirstSearchResult = firstElementOfSearchResults.getText();
+        new WebDriverWait(driver, 50).until(ExpectedConditions
+                .presenceOfAllElementsLocatedBy(By.xpath("//cite[contains(text(),'wikipedia')]")));
+        logger.info(String.format("First element of search results is %s", contentOfFirstSearchResult));
+            wait.until(pageLoadCondition);
```

Comment:
Did you format your code with Ctrl + Alt + L ?

Reply:
Yes, but it was before some refactoring, thank you for reminding.